### PR TITLE
Duplicate recipe

### LINKED ISSUE #124: [Story] Duplicate recipe
As an expert user, I can duplicate a recipe,so that I can save time for creating similar recipes.

### PR DESCRIPTION
**Pull request by StanleyW00**
Users are now able to duplicate a recipe based on the name or index.

Fixes #124

### CODE REVIEW

File: src/main/java/essenmakanan/command/DuplicateRecipeCommand.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        this.recipes = recipes;
+        this.toDuplicate = toDuplicate;
+    }
+
+    @Override
+    public void executeCommand() {
+        try {
+            int index = RecipeParser.getRecipeIndex(recipes, toDuplicate);
+            Recipe recipe = recipes.getRecipe(index);
+            Recipe copiedRecipe = new Recipe(recipe.getTitle() + " (copy)", recipe.getRecipeSteps()
```
Comment (kaijie0102) [inline src/main/java/essenmakanan/command/DuplicateRecipeCommand.java:25]: I like how you added a copy to indicate a duplicate

File: src/test/java/essenmakanan/recipe/DuplicateRecipeTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        String[] ingredients = {"i/bread,2,pc", "i/meat,1,pc"};
+        String[] steps = {"step1", "step2"};
+        recipe = new Recipe(recipeTitle, steps, ingredients);
+        recipes.addRecipe(recipe);
+    }
+
+    @Test
+    public void duplicateRecipe_numericalIndex_expectSimilarRecipe() {
+        DuplicateRecipeCommand command = new DuplicateRecipeCommand(recipes, "1");
+        command.executeCommand();
```
Comment (kaijie0102) [inline src/test/java/essenmakanan/recipe/DuplicateRecipeTest.java:30]: should this be in the DuplicateRecipeCommand test?

**Review by kaijie0102 — APPROVED** [review]
LGTM!

File: src/test/java/essenmakanan/recipe/DuplicateRecipeTest.java (hunk 2 above)
Comment (StanleyW00) [reply src/test/java/essenmakanan/recipe/DuplicateRecipeTest.java]: Sure, i will move it

**Pull request merged**